healthcareai/simple_mode.py

```diff
@@ -6,22 +6,52 @@ def __init__(self, dataframe, predicted_column, model_type, impute, grain_column
         self._dsm = DevelopSupervisedModel(dataframe, model_type, predicted_column, grain_column, verbose)
         self._dsm.data_preparation(impute=impute)
 
+    def random_forest(self):
+        # TODO Convenience method. Probably not needed?
+        if self._dsm.model_type is 'classification':
+            self.random_forest_classification()
+        elif self._dsm.model_type is 'regression':
+            self.random_forest_regression()
+
     def knn(self):
-        self._dsm.knn(
+        # Train the model
+        trained_model = self._dsm.knn(
             scoring_metric='roc_auc',
             hyperparameter_grid=None,
             randomized_search=True)
 
-    def random_forest(self):
-        if self._dsm.model_type is 'classification':
-            self._dsm.random_forest_classifier(
-                trees=200,
-                scoring_metric='roc_auc',
-                hyperparameter_grid=None,
-                randomized_search=True)
-        elif self._dsm.model_type is 'regression':
-            # TODO STUB
-            pass
+        # Display the model metrics
+        self.print_performance_metrics(trained_model)
+
+    def random_forest_regression(self):
+        # Train the model
+        trained_model = self._dsm.random_forest_regressor(trees=200, scoring_metric='roc_auc', randomized_search=True)
+        # Display the model metrics
+        self.print_performance_metrics(trained_model)
+
+    def random_forest_classification(self):
+        # Train the model
+        trained_model = self._dsm.random_forest_classifier(trees=200, scoring_metric='roc_auc', randomized_search=True)
+        # Display the model metrics
+        self.print_performance_metrics(trained_model)
+
+    def logistic_regression(self):
+        # Train the model
+        trained_model = self._dsm.logistic_regression()
+        # Display the model metrics
+        self.print_performance_metrics(trained_model)
+
+    def stochastic_gradient_descent_classifier(self):
+        # Train the model
+        trained_model = self._dsm.stochastic_gradient_descent_classifier()
+        # Display the model metrics
+        self.print_performance_metrics(trained_model)
```

Comment:
This code is looking really good and I'm excited about the progress.

I'm confused about stochastic_gradient_descent. In a CAFE chat, we had discussed that it was an optimization technique and not a representation (like random forest or logistic regression). These methods in this list right here should be similar in that they are all *representations*. See table 1 [here](https://homes.cs.washington.edu/~pedrod/papers/cacm12.pdf).